Data/HashMap/Internal.hs

```diff
@@ -579,23 +620,31 @@ isLeafOrCollision _               = False
 -- * Construction
 
 -- | /O(1)/ Construct an empty map.
-empty :: HashMap k v
-empty = Empty
+empty :: forall k v . HashMap k v
+empty = empty'
+
+-- | like 'empty' but allows a custom salt to be set
+empty' :: forall k v (salt :: Nat) . HashMapT salt k v
+empty' = Empty
 
 -- | /O(1)/ Construct a map with a single element.
-singleton :: (Hashable k) => k -> v -> HashMap k v
-singleton k v = Leaf (hash k) (L k v)
+singleton :: forall k v . (Hashable k) => k -> v -> HashMap k v
+singleton = singleton'
+
+-- | like 'singleton' but allows a custom salt to be set
+singleton' :: forall k v (salt :: Nat) . (Hashable k, KnownNat salt) => k -> v -> HashMapT salt k v
```

Comment:
Again, I think the salt should be the first type argument. That may even always be the case.